src/components/SignIn.tsx

```diff
@@ -326,8 +335,20 @@ const SignIn = ({ onWeb3ProviderSet, onWeb3AuthInstanceSet }: SignInProps) => {
   );
 
   useEffect(() => { setErrorMessage(null); }, [showSocialLogins, showMoreOptions]);
-
+  const getIconByName = (name: string) => {
+    switch(name) {
+      case 'MetaMask':
+        return <img src={iconMetamask} alt="metamask" />
+      case 'WalletConnect': 
+        return <img src={iconWalletConnect} alt="walletconnect" />
+      case 'Coinbase Wallet':
+        return <img src={iconCoinbase} alt="coinbase" />
+      default:
+        return <img src={iconTorus} alt="torus" />
+    }
```

Comment:
couple things here:
1. if you don't have default icon then you can just do basic object for these without `switch` and function
2. if you do want to have default icon then add one because now it defaults to `torus`

```suggestion
  const getIconByName = (name: string) => {
    switch(name) {
      case 'MetaMask':
        return <img src={iconMetamask} alt="metamask" />
      case 'WalletConnect': 
        return <img src={iconWalletConnect} alt="walletconnect" />
      case 'Coinbase Wallet':
        return <img src={iconCoinbase} alt="coinbase" />
      default:
        return <img src={iconTorus} alt="torus" />
    }
```

Reply:
We don't have a default icon now so that's why I kept the torus one as default, I'll connect with max and ccheck if we can have a default icon or not.

Reply:
Even with a default icon you could still use what Deimantas said, removing the switch statement
```suggestion
const iconsByName = { metamask: <icon />, coinbase: <icon />, ... };
const icon = iconByName[name] || iconByName['torus'];
```